concepts_5bb90f3b042863158cc7220c.md

```diff
@@ -51,17 +50,20 @@ We will see how to define and use secrets later in this guide. You can refer
 to the [secrets reference docs](https://docs.semaphoreci.com/article/51-secrets-yaml-reference)
 for more information.
 
-## Agents and Machine
+## Agents, Machines, and Containers
 
 Semaphore makes sure that there are always agents ready to run all your jobs.
 When configuring an agent, you can select from a number of [memory/CPU
-combinations][machine-types] and operating system environments.
+combinations][machine-types] and virtual machine (VM) operating system
+environments.
 
-The [Ubuntu Linux][ubuntu] and [MacOS Mojave][macos] environments have all
+The [Ubuntu Linux][ubuntu] and [MacOS Mojave][macos] VM environments have all
 the common build tools and programming languages preinstalled, so you can
-run your code without overhead.
-Agents also provide full `sudo` access to you so you can install additional
-software when needed.
+run your code without overhead. Agents also provide full `sudo` access to you
```

Comment:
- replace "all the common build tools" with "common build tools"
- Let's say "without an overhead of installing many dependencies in every workflow".